File: backend/app/types/sale.go
package types

type APISale struct {
	Datetime     string `json:"datetime"`
	Title        string `json:"title"`
	Image        string `json:"image"`
	Category     string `json:"category"`
	CategoryName string `json:"category_name"`
	Price        uint64 `json:"price"`
}

type APISales struct {
	Quantity int       `json:"quantity"`
	Sales    []APISale `json:"sales"`
}

File: frontend/src/types/sale.ts
export type SaleType = {
  datetime: string
  datetimeDisplay: boolean
  title: string
  image: string
  category: string
  category_name: string
  price: number
}

export type SalesType = {
  quantity: number
  sales: Array<SaleType>
}

export type SaleHistoryItemType = {
  image: string
  category_name: string
  title: string
  price: number
}

export type SaleHistoryType = Array<{
  datetime: string
  items: Array<SaleHistoryItemType>
}>
